Sources/PostgresNIO/New/PostgresChannelHandler.swift

```diff
@@ -407,51 +437,62 @@ final class PostgresChannelHandler: ChannelDuplexHandler {
                 context.writeAndFlush(self.wrapOutboundOut(self.encoder.flushBuffer()), promise: nil)
             }
             context.close(mode: .all, promise: promise)
+            if let promise {
+                promise.futureResult.cascade(to: runPromise)
+            } else {
+                runPromise?.succeed(())
+            }
```

Comment:
If the `closeConnection` has a promise in it, cascade the result of that to `runPromise`.
Other wise succeed `runPromise`.